### ISSUE TITLE
Downloading files with the same name

### ISSUE THREAD

**Issue by mckinsel** [dnanexus/dxWDL#148]
I'm running [this WDL](https://github.com/HumanCellAtlas/skylab/blob/master/pipelines/optimus/Optimus.wdl#L78-L85) with dxWDL, and I'm getting an error in the highlighted `MergeUnsorted` task. The error message is below, but it seems to be complaining that it can't localize both inputs files because they're both named "barcoded.bam".

I've tried recreating this in a simpler workflow a couple different ways, but so far I've only seen the error in this particular workflow. Any ideas?

```
Disambiguating file file-FGz8by003ZYkZvxkG9ZKfZQZ with name barcoded.bam
Eval env=Map(preemptible -> WomInteger(0), machine_mem_mb -> WomInteger(7500), disk -> WomInteger(500), docker -> WomString(us.gcr.io/broad-gotc-prod/genomes-in-the-cloud:2.3.3-1513176735), command_mem_mb -> WomInteger(7000), compression_level -> WomInteger(5), bam_inputs -> ["/home/dnanexus/execution/inputs/barcoded.bam", "/home/dnanexus/execution/inputs/1/barcoded.bam"], cpu -> WomInteger(1), sort_order -> WomString(unsorted))
calcInstanceType memory=Some(WomString(7500 MB)) disk=Some(WomString(local-disk 500 HDD)) cores=Some(WomInteger(1)) instancetype=mem1_ssd2_x8
required instance type: mem1_ssd2_x8
current instance type: mem1_ssd1_x4
isSufficient? false
Disambiguating file file-FGz8bXQ0gBVVkxG6G924q3fp with name barcoded.bam
java.nio.file.FileAlreadyExistsException: /home/dnanexus/execution/inputs/1/barcoded.bam```

**Comment by orodeh** [comment]
Well, the easiest thing to do is have files with different names.  

The generated applet is *supposed* to place input files with the same name in separate subdirectories (e.g, `inputs/1`). It looks like it isn't working due to a bug.

**Comment by orodeh** [comment]
This occurred due to faulty logic when you have three or more files with the same name, downloaded into a task.  Fixed in version 0.70.
